File: personal/abhinav/tmslt.cpp
// Codechef Practice Problems - Easy
// Team Split - TMSLT

#include <cstdio>
bool p[1000010],tmp;
int t;
long long i,n,a,b,c,d,rd;
int main()
{
  scanf("%d",&t);
  while(t--)
  {
    tmp=0;
    rd=0;
    scanf("%lld%lld%lld%lld%lld",&n,&a,&b,&c,&d);
    p[d]=1;
    for(i=1;i<n;i++)
    {
      d=((a*d+b)*d+c)%1000000;
      p[d]=!p[d];
    }
    for(i=999999;i>-1;i--)
      if(p[i])
      {
        if(tmp)
          rd-=i;
        else
          rd+=i;
        tmp = !tmp;
        p[i]=0;
      }
    printf("%lld\n",rd);
  }
  return 0;
}

File: personal/abhinav/tmslt.py
# Codechef Practice Problems - Easy
# Team Split - TMSLT

def doTheMagic(n,a,b,c,d):
    player_strength = [d]
    for i in range(1,n):
        player_strength.append((a*player_strength[i-1]*player_strength[i-1]\
            + b*player_strength[i-1] + c)%1000000)
    player_strength.sort()
    diff=0
    for i in range(n-1,-1,-2):
        diff += player_strength[i] - player_strength[i-1]
    return diff

t = int(input())
while(t):
    n,a,b,c,d = input().split(' ')
    print(doTheMagic(int(n),int(a),int(b),int(c),int(d)))
    t -= 1